### PR TITLE
Feat Followers

### PR DESCRIPTION
**Pull request by SanGeeky**
This PR adds the followers feature, the main changes are:

- Relation many-to-many between `users` and `followers`
- Add Follow and Unfollow users and their respective views
- Show followers and followed users
- Comments only for followers

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -9,6 +9,27 @@
     <span class="meta">
       <%= @user.first_name %> <%= @user.last_name %>
     </span>
+    <br>
+    <% if current_user %>
```
Comment (mauriciosabogal) [inline app/views/users/show.html.erb]: Avoid having too many levels of `if`s. You can refactor this code.

File: db/migrate/20210428151759_create_followers.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateFollowers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :followers do |t|
+      t.references :user, null: false, foreign_key: true
+      t.bigint :follower
```
Comment (mauriciosabogal) [inline db/migrate/20210428151759_create_followers.rb:5]: Why is `follower` not a foreign key?

File: app/models/user.rb [hunk 3]
```diff
@@ -4,8 +4,19 @@
...
+  has_many :follow_to,
+           class_name: :Follower,
+           foreign_key: :follower_id,
+           dependent: :destroy
 
   has_secure_password
   validates :email, presence: true, uniqueness: true
   validates :username, presence: true, uniqueness: true
+
+  def follow_user?(user_id)
```
Comment (mauriciosabogal) [inline app/models/user.rb:17]: This method returns a boolean if the user is the same, and an object if is not. The return type should be the same.

File: app/controllers/followers_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# followers_controller.rb
+class FollowersController < ApplicationController
+  before_action :authorized
+
+  def create
+    follow = Follower.create(follow_params)
+    follow.save
```
Comment (mauriciosabogal) [inline app/controllers/followers_controller.rb]: You are saving the object twice.

**Review by mauriciosabogal — CHANGES_REQUESTED** [review]

File: app/views/users/show.html.erb (hunk 1 above)
Comment (SanGeeky) [reply app/views/users/show.html.erb]: Done

File: app/controllers/followers_controller.rb (hunk 4 above)
Comment (SanGeeky) [reply app/controllers/followers_controller.rb]:
```
follow = Follower.new(follow_params)
follow.save if follow.valid?
redirect_article
```
Is true, now I am creating the object and saving

File: app/models/user.rb (hunk 3 above)
Comment (SanGeeky) [reply app/models/user.rb:17]:
```
 def follow_user?(user_id)
    following = follow_to.find_by(user_id: user_id)
 end
```
I refactor the conditional in view and the method

File: db/migrate/20210428151759_create_followers.rb (hunk 2 above)
Comment (SanGeeky) [reply db/migrate/20210428151759_create_followers.rb:5]: I only added for models now i add the migration for the database.
```
class AddForeignToFollowers < ActiveRecord::Migration[6.1]
  def change
    add_foreign_key :followers, :users, column: :follower_id
  end
end
```

**Review by mauriciosabogal — APPROVED** [review]

**Pull request merged**